webapp/channels/src/components/root/performance_reporter_controller.tsx

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useRef} from 'react';
+import {useStore} from 'react-redux';
+
+import {Client4} from 'mattermost-redux/client';
+
+import PerformanceReporter from 'utils/performance_telemetry/reporter';
+
+export default function PerformanceReporterController() {
+    const store = useStore();
+
+    const reporter = useRef<PerformanceReporter>();
+
+    useEffect(() => {
+        reporter.current = new PerformanceReporter(Client4, store);
+        reporter.current.observe();
+
+        // There's no way to clean up web-vitals, so continue to assume that this component won't ever be unmounted
+        return () => {
+            // eslint-disable-next-line no-console
+            console.error('PerformanceReporterController - Component unmounted or store changed');
+        };
+    }, [store]);
```

Comment:
when will store be changed? Is it on each store update ?

Reply:
Theoretically, it should never change unless we were to have to different stores for some reason like with `dispatch`. I only included it here because ESLint was complaining about it